website/src/components/Messages/MessageTableEntry.tsx

```diff
@@ -249,7 +250,6 @@ const MessageActions = ({
 
   const handleDelete = async () => {
     await deleteMessage();
-    mutate((key) => typeof key === "string" && key.startsWith("/api/messages"), undefined, { revalidate: true });
```

Comment:
You need to call mutate to revalidate the data here. The key should be something like this `/api/messages/${id}/tree`

Reply:
This is not needed at all if we properly pass the key to swr

Reply:
I now invalidate /api/messages/* and /api/admin/messages/* and /api/admin/user_messages